be/src/exprs/jit/jit_wrapper.cpp

```diff
@@ -0,0 +1,226 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exprs/jit/jit_wrapper.h"
+
+#include <glog/logging.h>
+
+#include <cassert>
+#include <iterator>
+#include <memory>
+#include <tuple>
+#include <utility>
+
+#include "common/compiler_util.h"
+#include "common/status.h"
+#include "exprs/jit/jit_functions.h"
+#include "llvm/ExecutionEngine/ExecutionEngine.h"
+#include "llvm/ExecutionEngine/Orc/LLJIT.h"
+#include "llvm/ExecutionEngine/Orc/SimpleRemoteEPC.h"
+#include "llvm/IR/DataLayout.h"
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/IR/Verifier.h"
+#include "llvm/MC/SubtargetFeature.h"
+#include "llvm/MC/TargetRegistry.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/Host.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/Transforms/InstCombine/InstCombine.h"
+#include "llvm/Transforms/Scalar.h"
+#include "llvm/Transforms/Scalar/GVN.h"
+#include "util/defer_op.h"
+
+namespace starrocks {
+
+Status JITWapper::init() {
+    if (_initialized) {
+        return Status::OK();
+    }
+
+    // Initialize LLVM targets and data layout.
+    llvm::InitializeNativeTarget();
+    llvm::InitializeNativeTargetAsmPrinter();
+    _target_machine = std::unique_ptr<llvm::TargetMachine>(llvm::EngineBuilder().selectTarget());
+    if (_target_machine == nullptr) {
+        LOG(ERROR) << "JIT: Failed to select target machine";
+        return Status::JitCompileError("Failed to select target machine");
+    }
+    _data_layout = std::make_unique<const llvm::DataLayout>(_target_machine->createDataLayout());
+
+    // Create a JIT engine instance.
+    auto jit = llvm::orc::LLJITBuilder().create();
+    if (!jit) {
+        std::string error_message;
+        llvm::handleAllErrors(jit.takeError(), [&](const llvm::ErrorInfoBase& EIB) { error_message = EIB.message(); });
+        LOG(ERROR) << "JIT: Failed to create LLJIT instance" << error_message;
+        return Status::JitCompileError("Failed to create LLJIT instance");
```

Comment:
error_message = "Failed to create LLJIT instance " + error_message;
LOG(ERROR) << error_message;
return Status::JitCompileError(error_message);

Reply:
Done.